src/code/TextTableParser.cs

```diff
@@ -0,0 +1,505 @@
+﻿// using System.Reflection.Metadata;
+using System.Diagnostics;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text.Json;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace TextTableParser
+{
+    public class ColumnInfo {
+        public int Start;
+        public int Length;
+        public int SpaceLength;
+    }
+
+    public class HeaderField {
+        public string Name;
+        public int Start;
+        public int Length;
+    }
+
+    public class ObjectPropertyInfo {
+        public HeaderField[] HeaderFields;
+        ObjectPropertyInfo(string line, ColumnInfo[] cInfos)
+        {
+            HeaderFields = new HeaderField[cInfos.Length];
+            for(int i = 0; i < cInfos.Length; i++)
+            {
+                HeaderFields[i] = new HeaderField(){
+                    Name = line.Substring(cInfos[i].Start, cInfos[i].Length).Trim(),
+                    Start = cInfos[i].Start,
+                    Length = cInfos[i].Length
+                };
+
+            }
+        }
+    }
+
+    [Cmdlet("Convert","TextTable")]
+    public class Class1 : PSCmdlet
+    {
+        [Parameter()]
+        public int MaximumWidth { get; set; } = 200;
+
+        [Parameter()]
+        public int WindowSize { get; set; } = 0; // 0 is all
+
+        [Parameter(ValueFromPipeline=true,Mandatory=true,Position=0)]
+        [AllowEmptyString()]
+        public string[]Line { get; set; }
+
+        [Parameter()]
+        public int Skip { get; set; } = 0;
+
+        [Parameter()]
+        public SwitchParameter NoHeader { get; set; }
+
+        [Parameter()]
+        public SwitchParameter AsPsObject { get; set; }

```

Comment:
I think it makes more sense to output PSObject by default and have `AsJson` to opt-in